app/models/host.rb

```diff
@@ -1770,4 +1770,22 @@ def validate_destroy
   def self.display_name(number = 1)
     n_('Host / Node', 'Hosts / Nodes', number)
   end
+
+  def self.post_refresh_ems(ems_id, _update_start_time)
+    ems = ExtManagementSystem.find(ems_id)
+    hosts = ems.hosts
+    unless hosts.empty?
+    hosts.find_each do |h|
+      begin
+        h.post_create_actions_queue
+      rescue NotImplementedError
+        _log.debug("Skipping Host [#{name}] id [#{id}], post_create_actions_queue not implemented in a subclass")
+      end
+    end
+    end
+  end
+
+  def post_create_actions_queue
```

Comment:
If you're trying to prevent these being queued for non-openstack hosts (good idea BTW) then you can just define the whole post_refresh_ems method in the openstack host model, you also need to set which classes to do post refresh for in the openstack refresher https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/base_manager/refresher.rb#L199-L202

Reply:
Oh I see you already did that https://github.com/ManageIQ/manageiq-providers-openstack/pull/380/files#diff-812a3c2ea73f9960e85e9f2f7908cc6aR16 :+1: